### PR TITLE
Feature/flux new api

### PR DESCRIPTION
**Pull request by koning**
Update to flux versions >= 0.48.x

### CODE REVIEW

**Comment by koning** [comment]
This include #405

File: merlin/examples/workflows/flux/scripts/flux_test/flux [hunk 1]
```diff
@@ -1,2 +1,8 @@
 #!/usr/bin/env python3
-print("Nodes")
+import sys
+
+
+if len(sys.argv) > 1 and sys.argv[1] == 'version':
+   print("commands:    		0.48.0")
```
[the remark this answers is not in the document]
Comment (koning) [reply merlin/examples/workflows/flux/scripts/flux_test/flux:6]: Do you mean the full output of a flux version call?

[the remark this answers is not in the document]
Comment (koning) [reply merlin/examples/workflows/flux/scripts/flux_test/flux:6]: This is what flux version prints:
``` 
node{koning}:flux version
commands:               0.47.0
libflux-core:		0.47.0
libflux-security:	0.8.0
build-options:		+hwloc==2.8.0+zmq==4.3.4
```

[the remark this answers is not in the document]
Comment (koning) [reply merlin/examples/workflows/flux/scripts/flux_test/flux:6]: I'll put in a string to make sure they know it is a fake flux.

[the remark this answers is not in the document]
Comment (doutriaux1) [reply merlin/examples/workflows/flux/scripts/flux_test/flux:6]: @koning why not print the actual version though?

[the remark this answers is not in the document]
Comment (bgunnar5) [reply merlin/examples/workflows/flux/scripts/flux_test/flux]: Would users download these test scripts if they run something like "merlin example flux"? Maybe we should move any scripts used strictly for testing to the testing directory to keep them separated

[the remark this answers is not in the document]
Comment (koning) [reply merlin/examples/workflows/flux/scripts/flux_test/flux]: Yes, I will move them to tests

**Review by bgunnar5 — APPROVED** [review]
LGTM. I'd like to get this merged before I put in a PR for the `batch.py` refactor so I can include your changes there as well.

**Comment by bgunnar5** [comment]
The pylint fixes have been merged so once you pull changes you're good to merge this whenever.

**Review by lucpeterson — APPROVED** [review]
Rock and roll

**Pull request merged**